tools/image_processing/imagej2/imagej2_bunwarpj_align.xml

```diff
@@ -1,75 +1,189 @@
-<?xml version='1.0' encoding='UTF-8'?>
 <tool id="imagej2_bunwarpj_align" name="Align two images" version="@WRAPPER_VERSION@.0">
     <description>with bUnwarpJ</description>
     <macros>
         <import>imagej2_macros.xml</import>
     </macros>
-    <expand macro="fiji_requirements" />
-    <command>
-<![CDATA[
-    python $__tool_directory__/imagej2_bunwarpj_align.py
-    --target_image "$target_image"
-    --target_image_format $target_image.ext
-    --source_image "$source_image"
-    --source_image_format $source_image.ext
-    #if $align_with_mask_cond.align_with_mask == 'yes':
-        --target_mask "$target_mask"
-        --target_mask_format $target_mask.ext
-        --source_mask "$source_mask"
-        --source_mask_format $source_mask.ext
-    #end if
-    --min_scale_def $min_scale_def
-    --max_scale_def $max_scale_def
-    --max_subsamp_fact $max_subsamp_fact
-    --divergence_weight $divergence_weight
-    --curl_weight $curl_weight
-    --image_weight $image_weight
-    --consistency_weight $consistency_weight
-    #if $advanced_options_cond.advanced_options == 'landmarks':
-        --landmarks_weight $advanced_options_cond.landmarks_weight
-        --landmarks_file "$advanced_options_cond.landmarks_file"
-    #else if $advanced_options_cond.advanced_options == 'affine':
-        --target_affine_file "$advanced_options_cond.target_affine_file"
-        --source_affine_file "$advanced_options_cond.source_affine_file"
-    #else if $advanced_options_cond.advanced_options == 'mono':
-        --mono "true"
-    #else if $advanced_options_cond.advanced_options == 'save_transformation':
-        --target_trans_out "$target_trans_out"
-        --source_trans_out "$source_trans_out"
+    <expand macro="fiji_requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+#import os
+#import tempfile
+#set error_log = 'output_log.txt'
+#set tmp_dir = $tempfile.mkdtemp(prefix='tmp-imagej-')
+
+## ImageJ2 requires file extensions to be valid image data types.
+#set source_out_suffix = '.' + str($source_out_datatype)
+#set target_out_suffix = '.' + str($target_out_datatype)
+#set tiff_suffix = '.tiff'
+#set source_image_sans_ext = $os.path.splitext($os.path.basename($source_image.file_name))[0]
+#set source_image_with_ext = '.'.join([$source_image_sans_ext, $source_image.ext])
+ln -s '$source_image.file_name' '$source_image_with_ext' &&
+#set target_image_sans_ext = $os.path.splitext($os.path.basename($target_image.file_name))[0]
+#set target_image_with_ext = '.'.join([$target_image_sans_ext, $target_image.ext])
+ln -s '$target_image.file_name' '$target_image_with_ext' &&
+
```

Comment:
@gregvonkuster can you explain me what you want to do here and how the filenames look before and after. I have the feeling that maybe we can do it easier with bash and using the CWD.